webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+    // Retrieve positions
+    String positions = request.getParameter("Positions");
+
+    int popup = 0;
+    if ((tempPopup != null) && (tempPopup.length() > 0)) {
+      popup = 1;
+    }
+
+    ArrayList<String> listIcons = new ArrayList<>();
+    if (listIcons != null) {
+      for (String icon : listeIcones.split(",")) {
+        if (!icon.isEmpty()) {
+          listIcons.add(icon);
+        }
+      }
+    }
+
+          /* recuperation de l'id */
+    String id = controller.getNextId();
+
+          /* Creer le repertoire id */
+    controller.createFolder(controller.getWebSitePathById(id));
+
+    // Persist siteDetail inside database type 0 = site cree
+    SiteDetail descriptionSite =
+        new SiteDetail(id, controller.getComponentId(), nomSite, description, nomPage, 0, null, null, 0,
+            popup);
+    descriptionSite.setPositions(positions);
+
+    String pubId = controller.createWebSite(descriptionSite);
+    descriptionSite = controller.getWebSite(id);
+    controller.setSessionSite(descriptionSite);
+
+    if (listIcons.size() > 0) {
+      controller.associateIcons(id, listIcons);
+    }
+
+    String currentPath = controller.getWebSitePathById(id);
+
+          /* ajout de la page principale */
+    String code = " ";
+
+          /* Creer la page principale */
+    controller.createFile(currentPath, nomPage, code);
+
+          /* publications : classer le site dans les themes cochés */
+    ArrayList<String> arrayToClassify = new ArrayList<String>();
+    boolean publish = false;
+    for (String idTopic : listeTopics.split(",")) {
+      if (!idTopic.isEmpty()) {
+        // ajout de la publication dans le theme
+        controller.addPublicationToFolder(pubId, idTopic);
+        publish = true;
+      }
+    }
+
+    if (publish) {
+      arrayToClassify.add(id);
+      controller.publish(arrayToClassify); // set etat du site a 1
+    }
+
+    request.setAttribute("Site", descriptionSite);
+    return currentPath;
+  }
+
+  private List<String> buildTab(String deb) {
+    /* deb = id/rep/ ou id\rep/ */
+    /* res = [id | rep] */
+    String node = "";
+    List<String> array = new ArrayList<>();
+    for(int i = 0; i < deb.length(); i++) {
+      char car = deb.charAt(i);
+      if (car == '/' || car == '\\') {
+        array.add(node);
+        node = "";
+      } else {
+        node += car;
+      }
+    }
+    return array;
+  }
+
+  private String getMachine(HttpServletRequest request) {
+    SettingBundle settings =
+        ResourceLocator.getSettingBundle("org.silverpeas.webSites.settings.webSiteSettings");
+    SettingBundle generalSettings = ResourceLocator.getGeneralSettingBundle();
+
+    StringBuilder machine = new StringBuilder(settings.getString("Machine", "").trim());
+    String context = (generalSettings.getString("ApplicationURL")).substring(1);
+    if (machine.length() == 0) {
+      List<String> a = buildTab(request.getRequestURL().toString());
+      int j = 1;
+      while (true) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS135)